src/routes/session.ts

```diff
@@ -1,30 +1,78 @@
+import * as _ from 'lodash';
 import {
 	loginUserXHR,
 	findUser,
 	getUser,
 	comparePassword,
 	runInvalidPasswordComparison,
 } from '../platform/auth';
-import { sbvrUtils } from '../platform';
+import { User as DbUser } from '../models';
+import { resinApi, root, sbvrUtils } from '../platform';
 import { captureException, handleHttpErrors } from '../platform/errors';
 import { resetCounter } from '../lib/rate-limiting';
 import { RequestHandler } from 'express';
 import { SetupOptions } from '..';
 
 const { BadRequestError, NotFoundError } = sbvrUtils;
 
-export const whoami: RequestHandler = (req, res) =>
-	getUser(req)
-		.then(user =>
-			res.send({ id: user.id, username: user.username, email: user.email }),
-		)
-		.catch(err => {
-			if (handleHttpErrors(req, res, err)) {
-				return;
+export const whoami: RequestHandler = async (req, res) => {
+	const userFields = ['id', 'username', 'email'];
```

Comment:
How about
```suggestion
	const userFields = ['id', 'username', 'email'] as const;
```
so that `pick` gives something better than just a `Partial<User>`?
PS: I'm not sure whether the `User` typings are correct on the `email` being non-nullable.